src/Microsoft.VisualBasic/tests/ConversionsTests.cs

```diff
@@ -2357,12 +2540,18 @@ public static IEnumerable<object[]> ToChar_InvalidObject_TestData()
 
         [Theory]
         [MemberData(nameof(ToChar_InvalidObject_TestData))]
-        [MemberData(nameof(InvalidBool_TestData))]
         public void ToChar_InvalidObject_ThrowsInvalidCastException(object value)
         {
             Assert.Throws<InvalidCastException>(() => Conversions.ToChar(value));
         }
 
+        [Theory]
+        [MemberData(nameof(InvalidBool_TestData))]
+        public void ToChar_InvalidBool_ThrowsInvalidOperationException(object value)
+        {
+            Assert.Throws<InvalidOperationException>(() => Conversions.ToChar(value));
+        }
+
         private static object s_floatEnum;
 
         public static object FloatEnum
```

Comment:
You can't make a float enum in VB or C#. Is it worth bothering to test this scenario? Ref.Emit is nice to avoid.

Reply:
Same for bool and char. I would just skip them.

Reply:
These edge cases used to throw a different exception in .NET Core compared to .NET Framework or threw when netfx didn't, so I would advocate keeping them. No harm really and there are already tests dotted around the place that use IntPtr/float etc. enums

Reply:
Ok